llmfoundry/models/layers/dmoe.py

```diff
@@ -66,11 +72,12 @@ def jitter(self, x: torch.Tensor) -> torch.Tensor:
         )
         return low + noise * (high - low)
 
-    def _top_k(self, scores: torch.Tensor) -> torch.Tensor:
+    def _top_k(self, scores: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
         if self.moe_top_k == 1:
-            return scores.max(
+            values, indices = scores.max(
                 dim=-1,
             )  # pyright: ignore[reportGeneralTypeIssues]
+            return values.unsqueeze(-1), indices.unsqueeze(-1)
```

Comment:
should you always unsqueeze?

Reply:
Yes, since the input to `_topk`, the `scores` matrix, is always 2D ([link](https://github.com/mosaicml/llm-foundry/blob/8274c6cf0e756816c7dc303370b68c7bb1663b27/llmfoundry/models/layers/dmoe.py#L84)). So the output of max gives 1D arrays for both `values` and `indices` as topk is 1. We want them to be 2D arrays